pallets/gear/src/tests.rs

```diff
@@ -417,6 +417,242 @@ fn restrict_start_section() {
+
+        ;; alloc 2nd page
+        (block
+            i32.const 1
+            call $alloc
+            i32.const 2
+            i32.eq
+            br_if 0
+            unreachable
+        )
+        ;; We freed 2nd page in init, so data will be default
+        (block
+            i32.const 0x20001
+            i32.load
+            i32.eqz
+            br_if 0
+            unreachable
+        )
+        ;; change 2nd page data
+        i32.const 0x20001
+        i32.const 0x42
+        i32.store
+        ;; free 2nd page
+        i32.const 2
+        call $free
+        ;; alloc it again
+        (block
+            i32.const 1
+            call $alloc
+            i32.const 2
+            i32.eq
+            br_if 0
+            unreachable
+        )
+        ;; write the same value
+        i32.const 0x20001
+        i32.const 0x42
+        i32.store
+
+        ;; 3th page. We have not access it yet, so data will be default
+        (block
+            i32.const 0x30001
+            i32.load
+            i32.eqz
+            br_if 0
+            unreachable
+        )
+        ;; change 3th page data
+        i32.const 0x30001
+        i32.const 0x42
+        i32.store
+        ;; free 3th page
+        i32.const 3
+        call $free
+        ;; then alloc it again
+        (block
+            i32.const 1
+            call $alloc
+            i32.const 3
+            i32.eq
+            br_if 0
+            unreachable
+        )
+        ;; write the same value
+        i32.const 0x30001
+        i32.const 0x42
+        i32.store
+
+        ;; 1st page. We have accessed this page before
+        (block
+            i32.const 0x10001
+            i32.load
+            i32.const 0x42
+            i32.eq
+            br_if 0
+            unreachable
+        )
+        ;; change 1st page data
+        i32.const 0x10001
+        i32.const 0x142
+        i32.store
+        ;; free 1st page
+        i32.const 1
+        call $free
+        ;; then alloc it again
+        (block
+            i32.const 1
+            call $alloc
+            i32.const 1
+            i32.eq
+            br_if 0
+            unreachable
+        )
+        ;; write the same value
+        i32.const 0x10001
+        i32.const 0x142
+        i32.store
+
+        ;; set new handle case
+        i32.const 0x0
+        i32.const 0x1
+        i32.store
+    )
+)
+"#;
+
+    init_logger();
+    new_test_ext().execute_with(|| {
+        let pid = {
+            let code = ProgramCodeKind::Custom(wat).to_bytes();
+            let salt = DEFAULT_SALT.to_vec();
+            let prog_id = generate_program_id(&code, &salt);
+            let res = GearPallet::<Test>::submit_program(
+                Origin::signed(USER_1),
+                code,
+                salt,
+                EMPTY_PAYLOAD.to_vec(),
+                500_000_000,
+                0,
+            )
+            .map(|_| prog_id);
+            res.expect("submit result was asserted")
+        };
```

Comment:
```suggestion
        let code = ProgramCodeKind::Custom(wat).to_bytes()
        let salt = DEFAULT_SALT.to_vec();
        let prog_id = generate_program_id(&code, &salt);
        assert_ok!(
            GearPallet::<Test>::submit_program(
                Origin::signed(USER_1),
                code,
                salt,
                EMPTY_PAYLOAD.to_vec(),
                500_000_000,
                0,
            )
        );
```

Reply:
I use submit program Id result to get pid

Reply:
But make some changes

Reply:
you see, mapping result and then expecting that is unnecessary.

Reply:
No, in such code I have more possibility to insert some loging in case something will goes wrong.